spec/registry.md

```diff
@@ -384,6 +322,166 @@ its _resolved value_ contains the implementation-defined integer value
+
+Some options do not have default values defined in this specification.
+The defaults for these options are implementation-dependent.
+In general, the default values for such options depend on the locale, 
+the currency,
+the value of other options, or all of these.
+
+Fraction digits for currency values behave differently than for other numeric formatters.
+The number of fraction digits displayed is usually set by the currency used.
+For example, USD uses 2 fraction digits, while JPY uses none.
+Setting some other number of `fractionDigits` allows greater precision display
+(such as when performing currency conversions or other specialized operations)
+or disabling fraction digits if set to `0`.
+The special _option_ _value_ `hideIfWhole` is used to display values without
+fraction digits when the number of fraction digits is zero,
+or based on the currency when the number of fraction digits for the currency is non-zero.
+> For example, this _message_:
+> ```
+> The special price is {$price :currency fractionDigits=hideIfWhole}.
+> ```
+> When used with the value `5.00 USD` in the `en-US` locale displays as:
+> ```
+> The special price is $5.
+> ```
+> But like this when when value is `5.01 USD`:
+> ```
+> The special price is $5.01.
+> ```
+
+Implementations MAY internally alias option values that they do not have data or a backing implementation for.
+Notably, the `currencyDisplay` option has a rich set of values that mirrors developments in CLDR data.
+Some implementations might not be able to produce all of these formats for every currency.
+
+> [!NOTE]
+> Except where noted otherwise, the names of _options_ and their _values_ were derived from the
+> [options](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat/NumberFormat#options)
+> in JavaScript's `Intl.NumberFormat`.
+
+> [!NOTE]
+> The option `select` does not accept the value `ordinal` because selecting
+> currency values using ordinal rules makes no sense.
+
+The following options and their values are required to be available on the function `:currency`:
+- `select`
+   -  `plural` (default)
+   -  `exact`
+- `currency`
+   - well-formed [Unicode Currency Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeCurrencyIdentifier)
+     (no default)
+- `compactDisplay` (this option only has meaning when combined with the option `notation=compact`)
+   - `short` (default)
+   - `long`
+- `notation`
+   - `standard` (default)
+   - `compact`
+- `numberingSystem`
+   - valid [Unicode Number System Identifier](https://cldr-smoke.unicode.org/spec/main/ldml/tr35.html#UnicodeNumberSystemIdentifier)
+     (default is locale-specific)
+- `currencySign`
+  - `accounting`
+  - `standard` (default)
+- `currencyDisplay` (this option's values are derived from those in ICU [NumberFormatter.UnitWidth](https://unicode-org.github.io/icu-docs/apidoc/released/icu4j/com/ibm/icu/number/NumberFormatter.UnitWidth.html))
+  - `narrow`
```

Comment:
@aphillips comments on this in https://github.com/unicode-org/message-format-wg/pull/915#discussion_r1829540496 with:
> No need to be extra verbose

The need here is for consistency. We should not be editorialising our source materials like this, and creating unnecessary inconsistencies.

I would also argue that "narrowSymbol" is much more clearly narrower than "symbol", and so without looking up the meanings in some source this makes it clearer which of "US$" and "$" is which, as otherwise a reader could more easily presume that the former is a "narrow" name, and the latter the "symbol".

```suggestion
  - `narrowSymbol`
```

Reply:
I'll buy your argument and am committing your suggestion. I'll note that `narrowSymbol`/`symbol` don't necessarily produce `$` and `US$`---it really depends on the locale.